### PR TITLE
basic display tutorial

### PR DESCRIPTION
**Pull request by alexholdenmiller**
to view this in html (recommended)...
```bash
git fetch
git checkout basic_tut
cd docs
make html
open build/html/basic_tutorial.html
```

### CODE REVIEW

File: docs/source/basic_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+``act`` produces an action from the agent.
+
+Observations are structured as a python `dict` with the following fields:
+
+
+.. image:: _static/img/act-obs-dict.png
+    :width: 60 %
+
+
+All of these fields are technically optional, and each task is free to use them
```
Comment (jaseweston) [inline docs/source/basic_tutorial.rst]: "as appropriate" -- well within limits.  its good if they are similar to other tasks as possible somehow? i would maybe reword a little..

File: docs/source/basic_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+Teachers
+^^^^^^^^
+
+A Teacher is special type of agent. They also implement the ``act`` and ``observe``
+functions like any agent does, but they also keep track of metrics which they
+return via a ``report`` function, such as the number of questions they have posed
+or how many times those questions have been answered correctly.
+
+Datasets will implement a subclass of Teacher, providing functions which
```
Comment (jaseweston) [inline docs/source/basic_tutorial.rst]: "will implement"  or maybe "typically implement" ?

File: docs/source/basic_tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+each call.
+
+A simple world included in ParlAI, which all of our currently included tasks use,
+is the ``DialogPartnerWorld``. DialogPartnerWorld is initialized with two agents,
+and with each call to ``parley``, one exchange is done between the agents, in
+the following manner:
+
+.. code-block:: python
+
+    query = teacher.act()
```
Comment (jaseweston) [inline docs/source/basic_tutorial.rst]: its not actually like this any more, but we could leave like that?

File: docs/source/basic_tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    teacher.observe(reply)
+
+Another simple world we include is MultiAgentDialogWorld, which is similar
+but generalizes this to cycle between any number of agents in a round robin
+fashion.
+
+Advanced Worlds
+^^^^^^^^^^^^^^^
+
+We also include a few more advanced worlds: in particular, we include both a
```
Comment (jaseweston) [inline docs/source/basic_tutorial.rst]: I would call these "container worlds" as they wrap other worlds

File: docs/source/basic_tutorial.rst (hunk 3 above)
Comment (alexholdenmiller) [inline docs/source/basic_tutorial.rst]: right yeah I thought this was waaayyyy more clear and so made better sense for the tutorial

File: docs/source/basic_tutorial.rst (hunk 2 above)
Comment (alexholdenmiller) [inline docs/source/basic_tutorial.rst]: changing to typically

File: docs/source/basic_tutorial.rst (hunk 1 above)
Comment (alexholdenmiller) [inline docs/source/basic_tutorial.rst]: on it

File: docs/source/basic_tutorial.rst (hunk 4 above)
Comment (alexholdenmiller) [inline docs/source/basic_tutorial.rst]: good call
